### PR TITLE
Scalability of knative publisher

### PR DESCRIPTION
**Pull request by montaro**
See also  #4754

### CODE REVIEW

File: components/event-bus/internal/knative/util/kn-eventing.go [hunk 1]
```diff
@@ -264,7 +278,11 @@ func resendMessage(httpClient *http.Client, channel *evapisv1alpha1.Channel, hea
 				log.Printf("ERROR: resendMessage(): could not resend HTTP request: %v", err)
 				return err
 			}
-			defer func() { _ = res.Body.Close() }()
+			defer func() {
+				_ = res.Body.Close()
+				tran, _ := httpClient.Transport.(*http.Transport)
```
Comment (sayanh) [inline components/event-bus/internal/knative/util/kn-eventing.go]: I think `tran.CloseIdleConnections()` needs to be in a separate `defer` right after `httpClient.Do(req)`.
If `httpClient.Do(req)` returns an error then we wouldn't be closing idle connections.

File: components/event-bus/internal/knative/util/kn-eventing.go [hunk 2]
```diff
@@ -192,27 +201,30 @@ func (k *KnativeLib) UpdateSubscription(sub *evapisv1alpha1.Subscription) (*evap
...
+	res, err := k.httpClient.Do(req)
 	if err != nil {
 		log.Printf("ERROR: SendMessage(): could not send HTTP request: %v", err)
 		return err
 	}
 	defer func() {
 		_ = res.Body.Close()
+		tran, ok := k.httpClient.Transport.(*http.Transport)
+		if !ok {
+			return
```
Comment (sayanh) [inline components/event-bus/internal/knative/util/kn-eventing.go]: I think this is more readable:
```
if tran, ok := k.httpClient.Transport.(*http.Transport); ok {
  tran.CloseIdleConnections()
}
```

File: components/event-bus/internal/knative/util/kn-eventing.go [hunk 3]
```diff
@@ -245,6 +257,8 @@ func resendMessage(httpClient *http.Client, channel *evapisv1alpha1.Channel, hea
 	}
 	defer func() {
 		_ = res.Body.Close()
+		tran, _ := httpClient.Transport.(*http.Transport)
```
Comment (sayanh) [inline components/event-bus/internal/knative/util/kn-eventing.go]: If typecasting fails then this will panic. Let's use the `ok` pattern here.

File: components/event-bus/internal/knative/util/kn-eventing.go (hunk 2 above)
Comment (montaro) [reply components/event-bus/internal/knative/util/kn-eventing.go]: Honestly, the first one is more readable to me :-)

File: components/event-bus/internal/knative/util/kn-eventing.go (hunk 3 above)
Comment (montaro) [reply components/event-bus/internal/knative/util/kn-eventing.go]: @sayanh Thanks for the catch, this was missed in the refactor, will fix it!

File: components/event-bus/internal/knative/util/kn-eventing.go (hunk 1 above)
Comment (montaro) [reply components/event-bus/internal/knative/util/kn-eventing.go]: `defer()` should be called after `resendMessage()` exits, right?

File: components/event-bus/internal/knative/util/kn-eventing.go (hunk 2 above)
Comment (montaro) [reply components/event-bus/internal/knative/util/kn-eventing.go]: I tried the other way but, I got a compilation error as `tran` would be unused var.

File: components/event-bus/internal/knative/util/kn-eventing.go (hunk 1 above)
Comment (montaro) [reply components/event-bus/internal/knative/util/kn-eventing.go]: fixed

File: components/event-bus/internal/knative/util/kn-eventing.go (hunk 2 above)
Comment (montaro) [reply components/event-bus/internal/knative/util/kn-eventing.go]: fixed as agreed

**Comment by montaro** [comment]
```
root@my-shell-65ffc94dc4-4gvnb /# ab -n 5000 -c 100 -p data -T application/json http://10.0.23.71:8080/v2/events
This is ApacheBench, Version 2.3 <$Revision: 1807734 $>
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/
Licensed to The Apache Software Foundation, http://www.apache.org/

Benchmarking 10.0.23.71 (be patient)
Completed 500 requests
Completed 1000 requests
Completed 1500 requests
Completed 2000 requests
Completed 2500 requests
Completed 3000 requests
Completed 3500 requests
Completed 4000 requests
Completed 4500 requests
Completed 5000 requests
Finished 5000 requests


Server Software:
Server Hostname:        10.0.23.71
Server Port:            8080

Document Path:          /v2/events
Document Length:        130 bytes

Concurrency Level:      100
Time taken for tests:   499.097 seconds
Complete requests:      5000
Failed requests:        0
Total transferred:      1195000 bytes
Total body sent:        1795000
HTML transferred:       650000 bytes
Requests per second:    10.02 [#/sec] (mean)
Time per request:       9981.948 [ms] (mean)
Time per request:       99.819 [ms] (mean, across all concurrent requests)
Transfer rate:          2.34 [Kbytes/sec] received
                        3.51 kb/s sent
                        5.85 kb/s total

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:        0    0   0.5      0       9
Processing:   376 9874 1061.7   9619   10865
Waiting:      376 9874 1061.7   9619   10865
Total:        379 9874 1061.4   9620   10865

Percentage of the requests served within a certain time (ms)
  50%   9620
  66%  10404
  75%  10580
  80%  10587
  90%  10595
  95%  10600
  98%  10630
  99%  10671
 100%  10865 (longest request)
```
image
image

File: resources/event-bus/values.yaml [hunk 4]
```diff
@@ -6,8 +6,8 @@ global:
   containerRegistry:
     path: eu.gcr.io/kyma-project
   event_bus:
-    dir: develop/
-    version: "36059183"
+    dir: pr/
```
Comment (sayanh) [inline resources/event-bus/values.yaml]: The changes in values.yaml can be in separate PR if you wanna avoid merging an image with PR tag.

Comment (montaro) [reply resources/event-bus/values.yaml]: I've seen folks doing following this approach recently
